src/Cassandra/Requests/PrepareHandler.cs

```diff
@@ -48,7 +50,7 @@ public PrepareHandler(ISerializer serializer, IInternalCluster cluster)
 
             if (session.Cluster.Configuration.QueryOptions.IsPrepareOnAllHosts())
             {
-                await SendRequestToAllNodesWithExistingConnections(session, request, prepareResult).ConfigureAwait(false);
+                await _reprepareHandler.ReprepareOnAllNodesWithExistingConnections(session, request, prepareResult).ConfigureAwait(false);
```

Comment:
Now that I think of it, we could get away without having a `PrepareResult` if we reuse `queryPlan` (it should continue iterating where it left it)

Reply:
I think that might be a step backwards, the end goal is to remove the LBP dependency from the prepare flow right? Right now the LBP is needed to select the first request only but I remember a discussion where we talked about selecting this host some other way without relying on the LBP.

Reply:
I think regardless of the origin of the query plan (it can be "not ignored" hosts with open connections to it, according to the distance all the lbps), the iterator is a better fit for "prepare one first" and "if it succeeds prepare on the rest", instead of storing additional information as the `PrepareResult`.

It's a suggestiion and it doesn't have to be implemented now as part of this fix.

Reply:
I think that's a good improvement but I'd rather separate that from this PR since that is something that shouldn't be merged to the DSE driver imo. Created https://datastax-oss.atlassian.net/browse/CSHARP-854